### PR TITLE
Improve performance of IsPrecededByWhitespace

### PR DESCRIPTION
**Pull request by pdelvo**
When analyzing performance the IsPrecededByWhitespace was using about ~3.5% total time when running our analyzers on Roslyn.sln.

After this change it is under 0.1%

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs [hunk 1]
```diff
@@ -64,14 +64,15 @@ internal static bool IsLastInLine(this SyntaxToken token)
...
+
+            if (pos < 0)
             {
-                return triviaList.Last().IsKind(SyntaxKind.WhitespaceTrivia);
+                return false;
             }
 
-            triviaList = token.GetPreviousToken().TrailingTrivia;
-            return triviaList.Count > 0 && triviaList.Last().IsKind(SyntaxKind.WhitespaceTrivia);
+            return char.IsWhiteSpace(token.SyntaxTree.GetText()[pos]);
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs]: :exclamation: `GetText(cancellationToken)`

:question: Should we handle the case where `SyntaxTree` is null (e.g. a token created by `SyntaxFactory.Token` that isn't in the tree yet)?

Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs]: Fixed

File: StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs [hunk 2]
```diff
@@ -61,17 +62,19 @@ internal static bool IsLastInLine(this SyntaxToken token)
         /// Gets a value indicating whether the <paramref name="token"/> is preceded by a whitespace.
         /// </summary>
         /// <param name="token">The token to process.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/>.</param>
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs]: :bulb: Description should be:

> The cancellation token that the operation will observe.

Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TokenHelper.cs]: fixed

**Comment by sharwell** [comment]
:+1:

**Pull request merged**